src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -345,6 +338,37 @@ const StudentApplication = ({ userProfile, authentication }) => {
     }
   };
 
+  /**
+   * Callback for changes to off-campus program info
+   * @param {String} offCampusProgramValue The program that the applicant is doing an OC program for
+   * @param {Number} index The index of the applicant in the list
+   */
+  const handleOffCampusInputChange = (offCampusProgramValue, index) => {
+    if (index !== null && index >= 0) {
+      // Get the profile of the selected user
+      const applicant = applicants[index];
+      let newApplicant = {
+        ...applicant,
+        OffCampusProgram: offCampusProgramValue ?? applicant.OffCampusProgram,
+      };
+      // Error checking on the hallNameValue before modifying the newHallInfo object
+      setApplicants((prevApplicants) =>
+        // replace the element at index with the new hall info object
+        prevApplicants.map((prevApplicant, j) => {
+          if (j === index) {
+            return newApplicant;
+          } else {
+            return prevApplicant;
+          }
+        }),
```

Comment:
You can simplify this using the inline conditional syntax
```suggestion
        prevApplicants.map((prevApplicant, j) =>
          j === index ? newApplicant : prevApplicant,
        ),
```